api/static/app/main.js

```diff
@@ -505,12 +505,19 @@ if (!nobrowsercheck && !isChrome()) {
     // Periodically load data
     var REFRESH_TIME_MINUTES = 5;
     function fetchAndReschedule() {
+        // If data doesn't load in 30 secs, show connection warning
+        timeout_interval = setTimeout(function(){
+            document.getElementById('connection-warning').style.display = "block";
+        }, 30000);
         queue()
             .defer(d3.json, 'europe.topo.json')
             .defer(d3.json, ENDPOINT + '/v1/production')
             .defer(d3.json, ENDPOINT + '/v1/solar')
             .defer(d3.json, ENDPOINT + '/v1/wind')
-            .await(dataLoaded);
+            .await(function(){
+                clearInterval(timeout_interval);
```

Comment:
ok - and just to be sure: will the warning *immediately* disappear when data isLoaded (i.e. when the overlay is removed)?